qiskit/pulse/schedule.py

```diff
@@ -141,17 +141,39 @@ def _instructions(self, time: int = 0) -> Iterable[Tuple[int, 'Instruction']]:
         for insert_time, child_sched in self._children:
             yield from child_sched._instructions(time + insert_time)
 
-    def union(self, *schedules: List[ScheduleComponent],
+    def union(self, *schedules: Union[ScheduleComponent, Tuple[int, ScheduleComponent]],
               name: Optional[str] = None) -> 'Schedule':
-        """Return a new schedule which is the union of `self` and `schedule`.
+        """Return a new schedule which is the union of children of both `self` and `schedules`.
 
         Args:
             *schedules: Schedules to be take the union with this `Schedule`.
             name: Name of the new schedule. Defaults to name of self
         """
         if name is None:
             name = self.name
-        return Schedule(self, *schedules, name=name)
+        new_sched = Schedule(name=name)
+        new_sched._union((0, self))
+        for sched_pair in schedules:
+            if not isinstance(sched_pair, tuple):
+                sched_pair = (0, sched_pair)
+            new_sched._union(sched_pair)
+        return new_sched
+
+    def _union(self, other: Tuple[int, ScheduleComponent]) -> 'Schedule':
+        """Mutably union `self` and `other` Schedule with shift time.
+
+        Args:
+            other: Schedule with shift time to be take the union with this `Schedule`.
+        """
+        shift_time, sched = other
+        sched_timeslots = sched.timeslots
+        shifted_children = sched._children
+        if shift_time != 0:
+            sched_timeslots = sched_timeslots.shift(shift_time)
+            shifted_children = [(t+shift_time, child) for t, child in shifted_children]
+        self._timeslots = self.timeslots.merged(sched_timeslots)
+        self.__children += tuple(shifted_children if shifted_children else [other])
```

Comment:
Where does `other` get added to children, if `this` has `._children`? I'm also not totally sure, but I think each item in `children` should be a `(time, Schedule)` tuple, so we are missing `time` here, right?

Reply:
Each item in children is a (time, Schedule) tuple as you expected, and  both `shifted_children` and `[other]` are list of `(time, Schedule)`, so I think we are not missing `time` here.

Reply:
@lcapelluto I made those lines easy to read at 337da93